crates/win-api-wrappers/src/wts.rs

```diff
@@ -229,30 +232,26 @@ pub fn send_message_to_session(
 }
 
 fn query_session_information_string(session_id: u32, info_class: WTS_INFO_CLASS) -> anyhow::Result<String> {
-    let mut string_ptr = windows::core::PWSTR::null();
+    use windows::core::PWSTR;
+
+    let mut string = PWSTR::null();
     let mut len: u32 = 0u32;
 
     matches!(info_class, WTSUserName | WTSDomainName)
         .then(|| 0)
         .ok_or_else(|| anyhow!("info_class is an unsupported WTS_INFO_CLASS: {}", info_class.0))?;
 
-    // SAFETY: All buffers are valid.
-    // `string_ptr` must be freed by `WTSFreeMemory`.
-    // Passing a `PWSTR` is correct only in the case that `info_class` represents a string result
+    // SAFETY: Passing a `PWSTR` is correct in the case where `info_class` represents a string result.
     unsafe {
-        WTSQuerySessionInformationW(
-            WTS_CURRENT_SERVER_HANDLE,
-            session_id,
-            info_class,
-            &mut string_ptr,
-            &mut len,
-        )?;
+        WTSQuerySessionInformationW(WTS_CURRENT_SERVER_HANDLE, session_id, info_class, &mut string, &mut len)?;
     }
 
-    // SAFETY: `string_ptr` is always a valid pointer on success.
-    let _ = unsafe { WTSMemory::from_raw(string_ptr.0.cast()) };
+    // SAFETY: On success, WTSQuerySessionInformationW places a valid pointer which must be freed by `WTSFreeMemory`.
+    let _guard = unsafe { WtsMemory::from_raw(string.as_ptr()) };
```

Comment:
`let _ = ` will cause the value to be immediately dropped.

See in [Rust Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=6c1de0468ef41093c81aa61658052861)

Reply:
Wow, I didn't know that, you live and learn... 😅 IMHO, that **huge** difference on such implicit thing gives me C++ vibes 🤔

Reply:
Yeah, the lint definitely would be useful in such cases